man/fi_peer.3.md

```diff
@@ -514,20 +545,25 @@ application has posted the matching receive buffer.
 
 ```
 1. A message is received by the peer provider.
-2. The peer calls owner->get_msg() / get_tag().
+2. The peer calls owner->get_msg() / get_tag(). If the incoming address is
+   FI_ADDR_UNSPEC, the owner cannot match this message to a receive posted with
+   FI_DIRECTED_RECV and can only match to receives posted with FI_ADDR_UNSPEC.
 3. The owner fails to find a matching receive buffer.
 4. The owner allocates a rx_entry with any known fields and returns -FI_ENOENT.
 5. The peer allocates any resources needed to handle the asynchronous processing
-   and sets peer_context accordingly.
-6. The peer allocates any needed resources for processing the unexpected
-   message and sets the peer_context accordingly, calling the owner's queue
-   function when ready to queue the unexpected message from the peer.
+   and sets peer_context accordingly, calling the owner's queue
+   function if the address is known or queue_progress if the address is
+   FI_ADDR_UNSPEC when ready to queue the unexpected message from the peer.
 7. The application calls fi_recv() / fi_trecv() on owner, posting the
    matching receive buffer.
-8. The owner matches the receive with the queued message on the peer.
-9. The owner removes the queued request, fills in the rest of the known fields
+8. The owner iterates through all queued progress entries and calls the peer's
```

Comment:
I'd like to see how the `progress()` is implemented. Is there a link to how it is handled in tcp provider? I can search it too.

Reply:
Nvm, find it in https://github.com/ofiwg/libfabric/pull/8907/commits/914c979b0a5af1b3260ef9e18697553b4ec85f0b#diff-8b61228cbd6da6eedda142bb8d2e0318c31db45b90cea54e091af319fa4d0f5e for util srx